midscribe-core/src/main/java/com/evolveum/midscribe/generator/PdfExporter.java

```diff
@@ -22,20 +15,38 @@ public String getDefaultExtension() {
     }
 
     @Override
-    public void export(File adocFile, File output) throws IOException {
-        File dir = output.getAbsoluteFile().getParentFile();
-        File file = new File(output.getName());
+    public void export(File adocFile, File outputPdf) throws IOException {
+        File tempHtmlFile = File.createTempFile("temp", ".html");
+
+        try {
+            HtmlExporter htmlExporter = new HtmlExporter();
+            htmlExporter.export(adocFile, tempHtmlFile);
+
+            String princeCommand = "prince " + tempHtmlFile.getAbsolutePath() + " -o " + outputPdf.getAbsolutePath();
```

Comment:
Why did you switch to external command? Asciidoctor supports export to PDF, this creates dependency on another binary that might not be available. Native PDF export via asciidoctor is preferred, in case prince is needed then I'd expected user will use html export and then prince command separately

Reply:
The main problem with the asciidoctor PDF export was formatting problem. Most of the generated documentation was unusable. The problem was that it did not recognize when the generated expression was ending therefore if the expression was longer and appeared on the next page it was overlapped with other sections. Use of the prince fixed that issue.

Reply:
@matejglombik I understand but can't accept 3rd party dependency this way. If I understand it correctly prince can be still used with `midscribe --output-format html --output document.html;` and then `prince document.html document.pdf` . We can rather try to fix pdf generator later to get rid of prince altogether.

Reply:
@matejglombik I've created some simple test with resource object that contains long groovy script (in terms of line count and line length), midscribe rendered pdf where script just truncated and didn't continue to next page, lines breaks were created correctly. Could you provide (send) xml that creates incorrect output please?